File: src/desktop/src-tauri/plugins/tauri-plugin-push/android/src/main/kotlin/com/alook/push/PushPlugin.kt
package com.alook.push

import android.app.Activity
import app.tauri.annotation.Command
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.JSObject
import app.tauri.plugin.Plugin
import app.tauri.plugin.Invoke
import com.google.firebase.messaging.FirebaseMessaging

@TauriPlugin
class PushPlugin(private val activity: Activity) : Plugin(activity) {

    private var pushToken: String? = null

    override fun load(webView: app.tauri.plugin.WebView) {
        super.load(webView)
        FirebaseMessaging.getInstance().token.addOnCompleteListener { task ->
            if (task.isSuccessful) {
                pushToken = task.result
                val event = JSObject()
                event.put("token", pushToken)
                event.put("platform", "android")
                trigger("token", event)
            }
        }
    }

    @Command
    fun getToken(invoke: Invoke) {
        val result = JSObject()
        if (pushToken != null) {
            result.put("token", pushToken)
            result.put("platform", "android")
            invoke.resolve(result)
        } else {
            invoke.reject("Push token not yet available")
        }
    }

    @Command
    fun onNotification(invoke: Invoke) {
        invoke.resolve()
    }
}

File: src/desktop/src-tauri/plugins/tauri-plugin-push/ios/Sources/PushPlugin.swift
import UIKit
import UserNotifications
import Tauri
import WebKit

class PushPlugin: Plugin {
    private var pushToken: String?

    override init() {
        super.init()
    }

    @objc override func load(webview: WKWebView) {
        super.load(webview: webview)
        requestPushPermission()
    }

    private func requestPushPermission() {
        UNUserNotificationCenter.current().requestAuthorization(options: [.alert, .badge, .sound]) { granted, _ in
            if granted {
                DispatchQueue.main.async {
                    UIApplication.shared.registerForRemoteNotifications()
                }
            }
        }
    }

    func didRegisterForRemoteNotifications(deviceToken: Data) {
        let token = deviceToken.map { String(format: "%02.2hhx", $0) }.joined()
        self.pushToken = token
        let event: [String: Any] = ["token": token, "platform": "ios"]
        self.trigger("token", data: event)
    }

    @objc func getToken(_ invoke: Invoke) {
        if let token = pushToken {
            invoke.resolve([
                "token": token,
                "platform": "ios"
            ])
        } else {
            invoke.reject("Push token not yet available")
        }
    }

    @objc func onNotification(_ invoke: Invoke) {
        invoke.resolve()
    }
}
